plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java

```diff
@@ -252,7 +287,43 @@ public void run() {
 				}
 				return projects;
 			}
+			
+			protected IService getService(Collection<IResource> resources) {
+				IResource service = getResourceOfType(resources, IService.class);
+				return service == null ? null : (IService)service;
+			}
+			
+			protected IRoute getRoute(Collection<IResource> resources) {
+				IResource route = getResourceOfType(resources, IRoute.class);
+				return route == null ? null : (IRoute)route;
+			}
+			
+			private IResource getResourceOfType(Collection<IResource> resources, Class<? extends IResource> type) {
+				for (IResource resource : resources) {
+					if (type.isInstance(resource)) {
```

Comment:
so we take the 1st available service? what if there are more services?
Furthermore we can create server adapters for replication controllers, deployment configs, too